receiver/githubreceiver/internal/webhook/config.go

```diff
@@ -0,0 +1,37 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package webhook // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/githubreceiver/internal/webhook"
+
+import (
+	"errors"
+
+	"go.opentelemetry.io/collector/component"
+	"go.uber.org/multierr"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/webhookeventreceiver"
+)
+
+var errMissingEndpointFromConfig = errors.New("missing receiver server endpoint from config")
+
+// Config defines configuration for GitHub Actions receiver
+type Config struct {
+	WebhookReceiver   webhookeventreceiver.Config `mapstructure:",squash"`             // squash ensures fields are correctly decoded in embedded struct
```

Comment:
see my comment about use of the webhookevent receiver as a library within this component.